url675964e2b9_pentaho_pentaho_commons_json_tgz-pJ8-org_json_TestJ8/src/org/json/JSONObject.java

```diff
@@ -0,0 +1,1490 @@
+     * For each of the methods with no parameters and a name starting
+     * with <code>"get"</code> or <code>"is"</code> followed by an uppercase letter,
+     * the method is invoked, and a key and the value returned from the getter method
+     * are put into the new JSONObject.
+     *
+     * The key is formed by removing the <code>"get"</code> or <code>"is"</code> prefix. If the second remaining
+     * character is not upper case, then the first
+     * character is converted to lower case.
+     *
+     * For example, if an object has a method named <code>"getName"</code>, and
+     * if the result of calling <code>object.getName()</code> is <code>"Larry Fine"</code>,
+     * then the JSONObject will contain <code>"name": "Larry Fine"</code>.
+     *
+     * @param bean An object that has getter methods that should be used
+     * to make a JSONObject.
+     */
+    public JSONObject(Object bean) {
+    	this();
+        populateInternalMap(bean, false);
+    }
+    
+    
+    /**
+     * Construct JSONObject from the given bean. This will also create JSONObject
+     * for all internal object (List, Map, Inner Objects) of the provided bean.
+     * 
+     * -- See Documentation of JSONObject(Object bean) also.
+     * 
+     * @param bean An object that has getter methods that should be used
+     * to make a JSONObject.
+     * @param includeSuperClass - Tell whether to include the super class properties.
+     */
+    public JSONObject(Object bean, boolean includeSuperClass) {
+    	this();
+        populateInternalMap(bean, includeSuperClass);
+    }
+    
+    private void populateInternalMap(Object bean, boolean includeSuperClass){
+    	Class klass = bean.getClass();
+    	
+        //If klass.getSuperClass is System class then includeSuperClass = false;
+    	
+    	if (klass.getClassLoader() == null) {
+    		includeSuperClass = false;
+    	}
+    	
+    	Method[] methods = (includeSuperClass) ? 
+    			klass.getMethods() : klass.getDeclaredMethods();
+        for (int i = 0; i < methods.length; i += 1) {
+            try {
+                Method method = methods[i];
+                String name = method.getName();
+                String key = "";
+                if (name.startsWith("get")) {
+                    key = name.substring(3);
+                } else if (name.startsWith("is")) {
+                    key = name.substring(2);
+                }
+                if (key.length() > 0 &&
+                        Character.isUpperCase(key.charAt(0)) &&
+                        method.getParameterTypes().length == 0) {
+                    if (key.length() == 1) {
+                        key = key.toLowerCase();
+                    } else if (!Character.isUpperCase(key.charAt(1))) {
+                        key = key.substring(0, 1).toLowerCase() +
+                            key.substring(1);
+                    }
+                    
+                    Object result = method.invoke(bean, (Object[])null);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Catch `InvocationTargetException` explicitly when you call `Method.invoke()`. Use `Throwable.getCause()` or `getTargetException()` in the catch handler to access more information about the underlying exception. See the following diffs on GitHub that make this code change: [diff1](https://github.com/EsfingeFramework/gamification/commit/20792047e6c69457cca03e877f2ada578188f32a#diff-6808c12fe9ddaea042db84f95dd5ba78adab99d0cb18bb4d85916a8723530768L21), [diff2](https://github.com/cucumber-attic/cuke4duke/commit/f4821d400cf2590126f1298a8dff9f7a33d45dc5#diff-80adaffc20c0e6e7d2e42312386ffe1ddc10213bf5b8e8919895fd85d666c8deL34)

Learn more: [InvocationTargetException in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/InvocationTargetException.html)